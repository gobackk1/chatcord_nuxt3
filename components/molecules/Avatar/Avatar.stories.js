import MolAvatar from './Avatar.vue'
import AtomImg from '../../atoms/Img/Img.vue'

export default {
  title: 'molecules/Avatar',
  component: MolAvatar,
  argTypes: {
    photoUrl: {
      description: '表示するプロフィール画像',
    },
    displayName: {
      description: 'ユーザーの表示名。alt属性値などに用いる',
    },
  },
  parameters: {
    docs: {
      description: {
        component:
          'v-avatarのラッパーコンポーネント。ユーザーがプロフィール画像を使用していない場合、プレースホルダーアイコンを表示する。',
      },
    },
  },
}

const Template = (args) => ({
  components: { MolAvatar, AtomImg },
  setup() {
    return { args }
  },
  template:
    '<mol-avatar :photo-url="args.photoUrl" :display-name="args.displayName" />',
})

export const Default = Template.bind({})
Default.args = {
  photoUrl: '',
  displayName: 'test user',
}
Default.parameters = {
  docs: {
    source: {
      code: '<mol-avatar :photo-url="null" display-name="test user" />',
    },
  },
}

export const WithPhotoURL = Template.bind({})
WithPhotoURL.args = {
  photoUrl: '/images/avatar.png',
  displayName: 'test user',
}
WithPhotoURL.parameters = {
  docs: {
    source: {
      code: '<mol-avatar photo-url="/images/avatar.png" display-name="test user" />',
    },
  },
}
