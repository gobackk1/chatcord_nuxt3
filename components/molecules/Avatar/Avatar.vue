<template>
  <v-avatar v-bind="$attrs">
    <atom-img v-if="photoUrl" :src="photoUrl" :alt="altText" />
    <template v-else
      ><v-icon size="x-large" :class="$style.icon"
        >mdi-account-circle</v-icon
      ></template
    >
  </v-avatar>
</template>

<script lang="ts">
import { defineComponent, PropType } from 'vue'
import AtomImg from '@/components/atoms/Img/Img.vue'

export default defineComponent({
  name: 'AppAvatar',
  components: { AtomImg },
  props: {
    photoUrl: {
      type: String as PropType<string | null | undefined>,
      required: true,
    },
    displayName: {
      type: String as PropType<string | null | undefined>,
      required: true,
    },
  },
  computed: {
    altText(): string {
      return `${this.displayName}のアバター画像`
    },
  },
})
</script>

<style module scss>
.icon {
  transform: scale(1.5);
}
</style>
