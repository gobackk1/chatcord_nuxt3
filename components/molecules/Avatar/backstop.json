{
  "scenarios": [
    {
      "label": "Default",
      "url": "http://localhost:6006/iframe.html?args=&id=molecules-avatar--default&viewMode=story",
      "misMatchThreshold" : 0.1
    },
    {
      "label": "WithPhotoURL",
      "url": "http://localhost:6006/iframe.html?args=&id=molecules-avatar--with-photo-url&viewMode=story",
      "misMatchThreshold" : 0.1
    }
  ]
}