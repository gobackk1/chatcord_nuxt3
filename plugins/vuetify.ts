import { createVuetify, type VuetifyOptions } from 'vuetify'
import 'vuetify/styles'
import { defineNuxtPlugin } from '#app'

export const vuetifyConfig: VuetifyOptions = {
  theme: {
    // defaultTheme: 'dark',
  },
  defaults: {
    VList: {
      density: 'compact',
    },
    VTextField: {
      density: 'compact',
      variant: 'outlined',
    },
    VAlert: {
      density: 'compact',
    },
  },
}

export default defineNuxtPlugin((nuxtApp) => {
  const vuetify = createVuetify(vuetifyConfig)
  nuxtApp.vueApp.use(vuetify)
})
