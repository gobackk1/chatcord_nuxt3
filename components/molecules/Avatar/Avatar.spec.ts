import { mount } from '@vue/test-utils'
import { createVuetify } from 'vuetify'
import Avatar from './Avatar.vue'

describe('UIコンポーネントのテスト', () => {
  let vuetify = createVuetify()
  beforeEach(() => {
    vuetify = createVuetify()
  })
  describe('ViewModelのテスト', () => {
    describe('ブラックボックステスト', () => {
      describe('同値・境界値テスト', () => {
        describe('displayName', () => {
          describe('有効同値', () => {
            describe('displayNameに「test user」と入力した場合', () => {
              it('デフォルトのアイコンを表示すること', () => {
                const wrapper = mount(Avatar, {
                  props: {
                    displayName: 'test user',
                  },
                  global: {
                    plugins: [vuetify],
                  },
                })
                expect(wrapper.find('i').classes()).toContain(
                  'mdi-account-circle'
                )
                expect(wrapper.html()).toMatchSnapshot()
              })
            })
          })
        })
        describe('photUrl, displayName', () => {
          describe('有効同値', () => {
            describe('photoUrlに「/path/to/photo.png」、displayNameに「test user」と入力した場合', () => {
              it('imgタグが描画されること', () => {
                const wrapper = mount(Avatar, {
                  props: {
                    displayName: 'test user',
                    photoUrl: '/path/to/photo.png',
                  },
                  global: {
                    plugins: [vuetify],
                  },
                })
                expect(wrapper.find('img').attributes().alt).toBe(
                  'test userのアバター画像'
                )
                expect(wrapper.find('img').attributes().src).toBe(
                  '/path/to/photo.png'
                )
                expect(wrapper.html()).toMatchSnapshot()
              })
            })
          })
        })
      })
    })
  })
})
