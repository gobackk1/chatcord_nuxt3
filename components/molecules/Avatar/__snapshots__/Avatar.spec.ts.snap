// Vitest Snapshot v1

exports[`UIコンポーネントのテスト > ViewModelのテスト > ブラックボックステスト > 同値・境界値テスト > displayName > 有効同値 > displayNameに「test user」と入力した場合 > デフォルトのアイコンを表示すること 1`] = `
"<div class=\\"v-avatar v-theme--light v-avatar--density-default v-avatar--size-default v-avatar--variant-flat\\"><i class=\\"mdi-account-circle mdi v-icon notranslate v-theme--light v-icon--size-x-large _icon_93c2a5\\" aria-hidden=\\"true\\"></i>
  <!----><span class=\\"v-avatar__underlay\\"></span>
</div>"
`;

exports[`UIコンポーネントのテスト > ViewModelのテスト > ブラックボックステスト > 同値・境界値テスト > photUrl, displayName > 有効同値 > photoUrlに「/path/to/photo.png」、displayNameに「test user」と入力した場合 >  1`] = `
"<div class=\\"v-avatar v-theme--light v-avatar--density-default v-avatar--size-default v-avatar--variant-flat\\"><img src=\\"/path/to/photo.png\\" alt=\\"test userのアバター画像\\">
  <!----><span class=\\"v-avatar__underlay\\"></span>
</div>"
`;

exports[`UIコンポーネントのテスト > ViewModelのテスト > ブラックボックステスト > 同値・境界値テスト > photUrl, displayName > 有効同値 > photoUrlに「/path/to/photo.png」、displayNameに「test user」と入力した場合 > imgタグが描画されること 1`] = `
"<div class=\\"v-avatar v-theme--light v-avatar--density-default v-avatar--size-default v-avatar--variant-flat\\"><img src=\\"/path/to/photo.png\\" alt=\\"test userのアバター画像\\">
  <!----><span class=\\"v-avatar__underlay\\"></span>
</div>"
`;
